types/fraction.go

```diff
@@ -0,0 +1,27 @@
+package types
+
+type Fraction struct {
+	Numerator   int64
+	Denominator int64
+}
+
+func (f *Fraction) Mul(m int64) Fraction {
+	return Fraction{f.Numerator * m, f.Denominator}
+}
+
+func (f Fraction) Int() int64 {
+	return f.Numerator / f.Denominator
+}
+
+type UFraction struct {
+	Numerator   uint64
+	Denominator uint64
+}
+
+func (f *UFraction) Mul(m uint64) UFraction {
+	return UFraction{f.Numerator * m, f.Denominator}
+}
+
+func (f UFraction) Int() uint64 {
+	return f.Numerator / f.Denominator
+}
```

Comment:
Whar about calling them `Floor` for both types? They return the greatest integer <= f.